docs/contribute/test/testing.md

```diff
@@ -0,0 +1,139 @@
+
+## Writing Tests for C++ Code Targeting WebAssembly
+
+Testing is a crucial part of software development, ensuring that code behaves as expected. In this guide, we'll explore how to write tests for C++ code specifically designed for WebAssembly (Wasm) environments, using Google Test (gtest) and a specific example related to the WasmEdge runtime.
+
+### Prerequisites
+- Familiarity with C++ and WebAssembly.
+- Basic understanding of Google Test (gtest) framework.
+- WasmEdge runtime environment set up for testing.
+
+### Code Explanation and Comments
+
+The provided code snippet demonstrates how to set up and execute tests for a C++ module targeting WebAssembly, specifically using the WasmEdge runtime. The code is structured into two main parts: the test setup and the test cases themselves.
+
+```cpp
+#include "common/defines.h"
+#include "runtime/instance/module.h"
+#include "wasi_logging/func.h"
+#include "wasi_logging/module.h"
+
+#include <gtest/gtest.h>
+#include <iostream>
+#include <spdlog/sinks/ostream_sink.h>
+#include <spdlog/spdlog.h>
+#include <sstream>
+
+namespace {
+
+// Function to create a module instance for testing.
+WasmEdge::Runtime::Instance::ModuleInstance *createModule() {
+  // ... Code omitted for brevity ...
+}
+
+// Helper function to fill memory with a byte value.
+void fillMemContent(WasmEdge::Runtime::Instance::MemoryInstance &MemInst,
+                    uint32_t Offset, uint32_t Cnt, uint8_t C = 0) noexcept {
+  // ... Code omitted for brevity ...
+}
+
+// Helper function to fill memory with a string.
+void fillMemContent(WasmEdge::Runtime::Instance::MemoryInstance &MemInst,
+                    uint32_t Offset, const std::string &Str) noexcept {
+  // ... Code omitted for brevity ...
+}
+
+} // namespace
+```
+
+### Test Cases
+
+```cpp
+TEST(WasiLoggingTests, func_log) {
+  // ... Code omitted for brevity ...
+
+  // Test cases for different logging levels and conditions.
+  // Each test case uses the `EXPECT_TRUE` or `EXPECT_FALSE` macros
+  // to validate the behavior of the `HostFuncInst.run` method.
+  
+  // ... Code omitted for brevity ...
+}
+```
+
+You can call the main function to run all the tests using the below given example
+
+```cpp
+GTEST_API_ int main(int argc, char **argv) {
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+```
+
+## Where to add the tests?
+
+After writing tests for your specific modules or plugins, you can add them under `https://github.com/WasmEdge/WasmEdge/tree/master/test` and for more specific placement of the tests, 
+
+For example,
+
+- You can add the tests for new plugins or update existing ones under `https://github.com/WasmEdge/WasmEdge/tree/master/test/plugins`
+
+## Documenting the Tests in Markdown
```

Comment:
Do we have such documents right now?

Reply:
Its for documenting, it does not say that we have such documents. Its for future devs to document their tests

Reply:
Where to put these files?

Reply:
Have added a note regarding the same